app/src/ui/ssh/ssh-user-password.tsx

```diff
@@ -37,9 +37,9 @@ export class SSHUserPassword extends React.Component<
         id="ssh-user-password"
         type="normal"
         title="SSH User Password"
-        dismissable={false}
+        backdropDismissable={false}
         onSubmit={this.onSubmit}
-        onDismissed={this.props.onDismissed}
+        onDismissed={this.onCancel}
```

Comment:
This made sure that when the dialog is dismissed via its built-in mechanisms (in this case, <kbd>Esc</kbd> and the close button), the input is actually cancelled and the process of the ongoing git operation is interrupted.